src/core/options-watcher.cpp

```diff
@@ -0,0 +1,134 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2023 Intel Corporation. All Rights Reserved.
+
+
+#include "options-watcher.h"
+#include <proc/synthetic-stream.h>
+
+
+namespace librealsense {
+
+
+options_watcher::options_watcher( std::chrono::milliseconds update_interval )
+    : _update_interval( update_interval )
+    , _updater()
+{
+}
+
+options_watcher::~options_watcher()
+{
+    stop();
+}
+
+void options_watcher::register_option( rs2_option id, std::shared_ptr< option > option, bool update_from_hw )
+{
+    _options[id] = option;
+    _update_from_hw[id] = update_from_hw;
+    _option_values[id] = update_from_hw ? option->query() : option->query(); // TODO - when implemented use option->get_last_known_value();
+
+    if( should_start() )
+        start();
+}
+
+void options_watcher::unregister_option( rs2_option id )
+{
+    _options.erase( id );
+
+    if( should_stop() )
+        stop();
+}
+
+rsutils::subscription
+options_watcher::subscribe( std::function< void( const std::map< rs2_option, std::shared_ptr< option > > & ) > && callback )
+{
+    rsutils::subscription ret = _on_values_changed.subscribe( std::move( callback ) );
+
+    if( should_start() )
+        start();
+
+    return ret;
+}
+
+rsutils::subscription_slot options_watcher::add_callback(
```

Comment:
Do we really need `add_callback` and `remove_callback`?

Reply:
Removed